TIVar.cpp

```diff
@@ -204,53 +211,60 @@ int TIVar::stringToStrVar8x(String s, uint8_t* strVar, enum Endpoint model) {
 		if (c < 0x20 || c >= 0x7f) {
 			// Ignore control characters and 8-bit codes
 			continue;
-		} else if ((c >= '0' && c <= '9') ||
-				   (c >= 'A' && c <= 'Z')) {
-			// Map basic characters (0-9, A-Z) directly
-			t = c;
-		} else if (c >= 'a' && c <= 'k') {
-			// Map lowercase letters (group 1)
-			t = c - 'a' + 0xbbb0;
-		} else if (c >= 'l' && c <= 'z') {
-			// Map lowercase letters (group 2)
-			t = c - 'l' + 0xbbbc;
-		} else {
-			// Map punctuation
-			switch (c) {
-				case ' ':	t = 0x29; break;
-				case '!':	t = 0x2d; break;
-				case '\"':	t = 0x2a; break;
-				case '#':	t = 0xbbd2; break;
-				case '$':	t = 0xbbd3; break;
-				case '%':	t = 0xbbda; break;
-				case '&':	t = 0xbbd4; break;
-				case '\'':	t = 0xae; break;
-				case '(':	t = 0x10; break;
-				case ')':	t = 0x11; break;
-				case '*':	t = 0x82; break;
-				case '+':	t = 0x70; break;
-				case ',':	t = 0x2b; break;
-				case '-':	t = 0x71; break;
-				case '.':	t = 0x3a; break;
-				case '/':	t = 0x83; break;
-				case ':':	t = 0x3e; break;
-				case ';':	t = 0xbbd6; break;
-				case '<':	t = 0x6b; break;
-				case '=':	t = 0x6a; break;
-				case '>':	t = 0x6c; break;
-				case '?':	t = 0xaf; break;
-				case '@':	t = 0xbbd1; break;
-				case '[':	t = 0x06; break;
-				case '\\':	t = 0xbbd7; break;
-				case ']':	t = 0x07; break;
-				case '^':	t = 0xf0; break;
-				case '_':	t = 0xbbd9; break;
-				case '`':	t = 0xbbd5; break;
-				case '{':	t = 0x08; break;
-				case '|':	t = 0xbbd8; break;
-				case '}':	t = 0x09; break;
-				case '~':	t = 0xbbcf; break;
+		}
+
+		if (type == STR_83) {
+			if ((c >= '0' && c <= '9') ||
+					   (c >= 'A' && c <= 'Z')) {
+				// Map basic characters (0-9, A-Z) directly
+				t = c;
+			} else if (c >= 'a' && c <= 'k') {
+				// Map lowercase letters (group 1)
+				t = c - 'a' + 0xbbb0;
+			} else if (c >= 'l' && c <= 'z') {
+				// Map lowercase letters (group 2)
+				t = c - 'l' + 0xbbbc;
+			} else {
+				// Map punctuation
+				switch (c) {
+					case ' ':	t = 0x29; break;
```

Comment:
I should really turn this into a tabular structure at some point, but let's call that future work.